ChildProject/pipelines/sampler.py

```diff
@@ -79,6 +80,86 @@ def add_parser(samplers):
         parser.add_argument('--target-speaker-type', help = 'speaker type to get chunks from', choices=['CHI', 'OCH', 'FEM', 'MAL'], nargs = '+', default = ['CHI'])
         parser.add_argument('--sample-size', help = 'how many samples per recording', required = True, type = int)
 
+import numpy as np
+from pydub import AudioSegment
+
+class EnergyDetectionSampler(Sampler):
+    def __init__(self,
+        project: ChildProject.projects.ChildProject,
+        windows_length: float,
+        windows_spacing: float,
+        windows_count: int,
+        threshold: float = None,
+        lowpass_freq: int = -1,
+        initial_segments: str = None
+        ):
+
+        super().__init__(project)
+        self.windows_length = windows_length
+        self.windows_count = windows_count
+        self.windows_spacing = windows_spacing
+        self.threshold = threshold
+        self.lowpass_freq = lowpass_freq
+        self.initial_segments_path = initial_segments
+
+    def compute_energy_loudness(self, chunk, sampling_frequency: int):
+        chunk_fft = np.fft.fft([x/(2**15 - 1) for x in chunk])
+
+        freq = np.fft.fftfreq(len(chunk_fft), 1/sampling_frequency)
+
+        if self.lowpass_freq > 0:
+            chunk_fft = chunk_fft[(freq < self.lowpass_freq)]
+        
+        return np.sum(np.abs(chunk_fft)**2)/len(chunk_fft)
+
+    def get_recording_windows(self, profile, recording):
+        recording_path = os.path.join(self.project.path, ChildProject.projects.ChildProject.RAW_RECORDINGS, recording['filename'])
+        audio = AudioSegment.from_wav(recording_path)
+        duration = audio.duration_seconds
+        channels = audio.channels
+        frequency = int(audio.frame_rate)
+
+        windows_starts = (1000*np.arange(self.windows_spacing, duration - self.windows_length, self.windows_spacing)).astype(int)
+        windows = []
+
+        for start in windows_starts:
+            energy = 0
+            chunk = audio[start:start+int(1000*self.windows_length)].get_array_of_samples()
+            
+            for channel in range(channels):
```

Comment:
So, this is how we currently handle multiple channels : we sum the energy across channels.
It's always more meaningful than summing amplitudes... But is it really ?